### DISCUSSION TITLE
Pvr offset management

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#24] [Ideas]
As described on the CE forum, having a specific progress bar calculation to manage pvr offset
https://discourse.coreelec.org/t/graphical-front-panel-display/12932/191

This would need a section in `setup.toml` file where you define something like
```
use_PVR_offset = true 
PVR_start_offset= 600 #time (in second) that PVR record before the recorded program
PVR_stop_offset=600 #time (in second) that PVR record after the recorded program
```

Also these offsets could be used to create a special treatment on  `VideoPlayer.Time` and `VideoPlayer.Duration` info labels to display correct time.

As these values could be negative, maybe a method to display negative value could be imagined (change of color, adding a '-' character, etc)

**Comment by mattblovell** [comment]
In the latest kodi_panel_display.py (for not-yet-released v1.27), I've modified *where* `calc_progress()` gets called.

That function call now occurs within either `audio_screens()` or `video_screens()` **after** any mode selection has occurred.  The `calc_progress()` function now gets passed the selected layout name (as a string).  So, one *could* replace the `calc_progress()` function.

However, per Post 202 on that same CoreELEC thread, the variability in PVR offset times that sounds possible seems like it reduces the utility of proceeding much further along this path.  *If* you have uniform offsets, then you could now write your own `calc_progress()` substitute function, identical to the original except if the layout_name string is "V_PVR".  If that layout is being used, the calculations can then be modified as desired.

The change to kodi_panel_display.py is in commit 4dcda63393fd0bdc0e95bd288bb76519141634a7.

**Comment by nico1080** [comment]
This solution work Ok.

I added my offset with this function: https://github.com/nico1080/kodi_panel/commit/b876444158cf4cffdc398a98b0aabe36927b5c58
